From a2db9fd9e2c67e57ecc1463ed3bdd047d6f7cc14 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Tue, 20 Aug 2019 14:36:38 +0530
Subject: [PATCH 0268/1921] octeontx2-af: Always enable mcam rules for TX

Enabling MCAM rule depends on whether the NIXLF
is initialized or not. MCAM rules for TX can
always be enabled because packet hitting a MCAM
rule in TX path implies that it was transmitted
by a initialized NIXLF.

Change-Id: I59f33ec042b852253240a3c29fec0136066ad6ec
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/14559
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
index 52f508083bfa..edbae9a7969a 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_npc_fs.c
@@ -1026,6 +1026,13 @@ int rvu_mbox_handler_npc_install_flow(struct rvu *rvu,
 	if (err || (!req->default_rule && !pfvf->def_rule))
 		enable = false;
 
+	/* Packets reaching NPC in Tx path implies that a
+	 * NIXLF is properly setup and transmitting.
+	 * Hence rules can be enabled for Tx.
+	 */
+	if (req->intf == NIX_INTF_TX)
+		enable = true;
+
 	/* Do not allow requests from uninitialized VFs */
 	if (from_vf && !enable)
 		return -EINVAL;
-- 
2.31.1

